lib/ui/pages/score/widgets/score_tile_widget.dart

```diff
@@ -0,0 +1,83 @@
+// ignore_for_file: import_of_legacy_library_into_null_safe
+
+import 'package:auto_size_text/auto_size_text.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_app/src/model/course/course_score_json.dart';
+
+typedef OnCategoryChanged = void Function(int? category);
+
+class ScoreTile extends StatelessWidget {
+  ScoreTile({
+    super.key,
+    required String courseName,
+    required String category,
+    required String scoreValue,
+    OnCategoryChanged? onCategoryChanged,
+  })  : _courseName = courseName,
+        _category = category,
+        _scoreValue = scoreValue,
+        _onCategoryChanged = onCategoryChanged;
+
+  /// The score value of a course.
+  /// Note that we should make the score's type to be a [String] instead of [int] since the score can be a string like "Q".
+  final String _scoreValue;
+  final String _category;
+  final String _courseName;
+  final OnCategoryChanged? _onCategoryChanged;
+
+  final ValueNotifier<int?> _selectedCategory = ValueNotifier(null);
+
+  int? _getInitialCategoryIndex() {
+    final index = constCourseType.indexOf(_category);
+    return index == -1 ? null : index;
+  }
+
+  Widget _buildCourseName() => AutoSizeText(
+        _courseName,
+        style: const TextStyle(fontSize: 16.0),
+      );
```

Comment:
Suggest changing to getter and rename variable.

```suggestion
  Widget get _courseNameText => AutoSizeText(
        _courseName,
        style: const TextStyle(fontSize: 16.0),
      );
```